api/app/signals/apps/api/v1/views/pdf.py

```diff
@@ -74,6 +81,49 @@ def get_object(self):
         self.check_object_permissions(request=self.request, obj=obj)
         return obj
 
+    def _resize(self, image):
+        # Consider image orientation:
+        if image.width > image.height:
+            # landscape
+            width = self.max_size
+            height = int((self.max_size / image.width) * image.height)
+        else:
+            # portrait
+            width = int((self.max_size / image.height) * image.width)
+            height = self.max_size
+
+        return image.resize(size=(width, height), resample=Image.LANCZOS).convert('RGB')
+
+    def _get_context_data_images(self, signal):
+        jpg_data_urls = []
+        for att in signal.attachments.all():
+            # Attachment is_image property is currently not reliable
+            _, ext = os.path.splitext(att.file.name)
+            if ext not in ['.gif', '.jpg', '.jpeg', '.png']:
+                continue  # unsupported image format, or not image format
+
+            try:
+                with default_storage.open(att.file.name) as file:
+                    buffer = io.BytesIO(file.read())
+                image = Image.open(buffer)
+            except (UnidentifiedImageError, FileNotFoundError):
+                continue  # Protect against missing files and bad data.
```

Comment:
Log the errors in Sentry.

Consider splitting this into 2 except blocks so that we can decide how to log it in Sentry. A missing file is more interesting (a real error) than "bad data" (perhaps a warning is enough here). Because "bad data" can also be a different kind of attachment, for example a PDF or a Word document.

Reply:
I changed this to catch the `UnidentifiedImageError` and only log that the file is unopenable. If some other problem occurs, a warning is emitted and the exception is included. We want the PDF to be output, so we skip the attachment that causes the problem.